src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -414,26 +331,15 @@ public bool TryGetProperty(ReadOnlySpan<byte> utf8PropertyName, out JsonElement
         /// </exception>
         public bool GetBoolean()
         {
-            CheckValidInstance();
-
-            if (_parent is JsonDocument document)
-            {
-                JsonTokenType type = TokenType;
-
-                return
-                    type == JsonTokenType.True ? true :
-                    type == JsonTokenType.False ? false :
-                    throw ThrowHelper.GetJsonElementWrongTypeException(nameof(Boolean), type);
-            }
+            // CheckValidInstance is redundant.  Asking for the type will
```

Comment:
@bartonjs, should we still keep this to validate the `JsonElement elem = default` cases?

Reply:
I was going to make the same comment, then checked and saw that this is just restoring it exactly to what shipped in 3.1, so it's good.